Gems/Atom/RHI/Code/Source/RHI/Factory.cpp

```diff
@@ -11,20 +11,33 @@
 #include <AzCore/Interface/Interface.h>
 #include <AzCore/Component/TickBus.h>
 
-#if defined(USE_RENDERDOC)
+#if defined(USE_RENDERDOC) || defined(USE_PIX)
 #include <AzCore/Module/DynamicModuleHandle.h>
 #include <Atom/RHI/RHIUtils.h>
 #include <Atom_RHI_Traits_Platform.h>
+#endif
 
+#if defined(USE_RENDERDOC)
 static AZStd::unique_ptr<AZ::DynamicModuleHandle> s_renderDocModule;
 static RENDERDOC_API_1_1_2* s_renderDocApi = nullptr;
+static bool s_isRenderDocDllLoaded = false;
```

Comment:
nit: why not define both flags within the class?

Reply:
Its because we can have multiple factories. One per RHI.